server/src/main/java/io/crate/planner/optimizer/rule/EliminateCrossJoin.java

```diff
@@ -171,7 +174,8 @@ static LogicalPlan reorder(JoinGraph graph, List<LogicalPlan> order) {
                         errorMessage.add(relationName.fqn());
                     }
                 }
-                throw new InvalidArgumentException("JoinPlan cannot be built with the provided order " + errorMessage);
+                LOGGER.trace("JoinPlan cannot be built with the provided order {}", errorMessage);
+                return null;
```

Comment:
If we fail on reordering, let's skip this rule by returning null.

Reply:
yep, I agree, but I think we should use: `LoggingOptimizerTracer` and not a specific logger for this class, wdyt?

Reply:
`OptimizerTracer` is used to track the progress of the optimizer not for error messages.

Reply:
So `LoggingOptimizerTracer` tells us which rule was executed and what progress/transformation was made on the query plan.

Reply:
ok, I see, makes sense to leave as is, thx!